internal/cli/atlas/accesslists/create.go

```diff
@@ -80,34 +87,87 @@ func (opts *CreateOpts) newProjectIPAccessList() []*atlas.ProjectIPAccessList {
 	return []*atlas.ProjectIPAccessList{entry}
 }
 
+func getIPAddress() string {
+	if publicIP := store.IPAddress(); publicIP != "" {
+		return publicIP
+	}
+
+	_, _ = fmt.Fprintln(os.Stderr, currentIPAddressNotFound)
+	return ""
+}
+
+func validateCurrentIPFlag(opts *CreateOpts, cmd *cobra.Command, args []string) (bool, error) {
+	needsArg := true
+	hasArgs := len(args) > 0
+
+	if opts.entryType == ipAddress && !hasArgs {
+		needsArg = false
+	}
+
+	if !needsArg && (!opts.currentIP && !hasArgs) || (opts.currentIP && hasArgs) {
+		return needsArg, fmt.Errorf(
+			"please either provide <entry> or use %s to use current IP Address.\n\nUsage: %s",
+			flag.CurrentIP,
+			cmd.UseLine(),
+		)
+	}
+
+	if needsArg && len(args) == 0 {
+		return needsArg, fmt.Errorf(
+			"%q with entry type %s requires at least %d argument received %d\n\nUsage:  %s",
+			cmd.CommandPath(),
+			opts.entryType,
+			1,
+			len(args),
+			cmd.UseLine(),
+		)
+	}
+
+	return needsArg, nil
```

Comment:
I may be missing something here

```suggestion
	if needsArg && !opts.currentIP {
		return fmt.Errorf(
			"please either provide <entry> or use %s to use current IP Address.\n\nUsage: %s",
			flag.CurrentIP,
			cmd.UseLine(),
       }

	return nil
```

Reply:
there are 2 things here: we need to send an error if the user passes `IP` and `--currentIP` flag and also if the command is of another type and we require an argument but it's empty (since I removed the `requireArgs`) . I'll still simplify the first if statement as I overcomplicated things